### PR TITLE
drivers: sensor: adxl345: implement interrupt support

### PR DESCRIPTION
**Pull request by badochov**
The adxl345 sensor driver was missing the interrupt support. Interrupt support simmilar to other accelerometers from its family was added.

### CODE REVIEW

File: drivers/sensor/adxl345/Kconfig [hunk 1]
```diff
@@ -11,3 +11,146 @@ config ADXL345
...
+	  Priority of thread used by the driver to handle interrupts.
+
+config ADXL345_THREAD_STACK_SIZE
+	int "Thread stack size"
+	depends on ADXL345_TRIGGER_OWN_THREAD && ADXL345_TRIGGER
+	default 1024
+	help
+	  Stack size of thread used by the driver to handle interrupts.
+
+config ADXL345_ACTIVITY_THRESHOLD
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/Kconfig:55]: This and most of the following configs should be moved to devicetree properties to allow different values per device instance.

File: drivers/sensor/adxl345/adxl345.c [hunk 2]
```diff
@@ -91,7 +91,6 @@ static inline int adxl345_reg_write_byte(const struct device *dev, uint8_t addr,
 }
 
 static inline int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
-
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/adxl345.c:94]: Revert the unreleated whitespace changes or move them to a separate commit

File: drivers/sensor/adxl345/adxl345.c [hunk 3]
```diff
@@ -240,49 +619,66 @@ static int adxl345_init(const struct device *dev)
 		return -EIO;
 	}
 
-	rc = adxl345_reg_write_byte(dev, ADXL345_RATE_REG, ADXL345_RATE_25HZ);
+	rc = adxl345_reg_write_byte(dev, ADXL345_RATE_REG, CONFIG_ADXL345_ODR & 0xf);
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/adxl345.c:622]: Is this related to adding interrupt support?

File: drivers/sensor/adxl345/adxl345.c [hunk 4]
```diff
@@ -240,49 +619,66 @@ static int adxl345_init(const struct device *dev)
...
+#define ADXL345_DEFINE(inst)                                                                       \
+	static struct adxl345_dev_data adxl345_data_##inst;                                        \
+                                                                                                   \
+	static const struct adxl345_dev_config adxl345_config_##inst =                             \
+		COND_CODE_1(DT_INST_ON_BUS(inst, spi), (ADXL345_CONFIG_SPI(inst)),                 \
+			    (ADXL345_CONFIG_I2C(inst)));                                           \
+                                                                                                   \
+	SENSOR_DEVICE_DT_INST_DEFINE(inst, adxl345_init, NULL, &adxl345_data_##inst,               \
+				     &adxl345_config_##inst, POST_KERNEL,                          \
+				     CONFIG_SENSOR_INIT_PRIORITY, &adxl345_api_funcs);
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/adxl345.c:682]: The whitespace changes add a lot of noise and make it difficult to see what functional changes are being made. It looks like you're just adding `ADXL345_CONFIG_INTERRUPT_PART(inst)` to the CONFIG_SPI and CONFIG_I2C macros? Can you just add those lines without the additional whitespace changes?

File: dts/bindings/sensor/adi,adxl345-i2c.yaml [hunk 5]
```diff
@@ -5,4 +5,4 @@ description: ADXL345 3-axis I2C accelerometer
 
 compatible: "adi,adxl345"
 
-include: [sensor-device.yaml, i2c-device.yaml]
+include: [sensor-device.yaml, i2c-device.yaml, "adi,adxl345-common.yaml"]
```
Comment (MaureenHelm) [inline dts/bindings/sensor/adi,adxl345-i2c.yaml:8]: You can remove `sensor-device.yaml` here since it's now being included via `adi,adxl345-common.yaml`

File: dts/bindings/sensor/adi,adxl345-spi.yaml [hunk 6]
```diff
@@ -5,4 +5,4 @@ description: ADXL345 3-axis accelerometer with SPI connection
 
 compatible: "adi,adxl345"
 
-include: [sensor-device.yaml, spi-device.yaml]
+include: [sensor-device.yaml, spi-device.yaml, "adi,adxl345-common.yaml"]
```
Comment (MaureenHelm) [inline dts/bindings/sensor/adi,adxl345-spi.yaml:8]: You can remove `sensor-device.yaml` here since it's now being included via `adi,adxl345-common.yaml`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for the contribution

File: drivers/sensor/adxl345/Kconfig (hunk 1 above)
Comment (whati001) [reply drivers/sensor/adxl345/Kconfig:55]: Hi, I am also currently dealing with the ADXL345 and have found this PR, which would enable what I need from the accel. It seems like all ADXL accel sensors leverage the KConfig system to define the ACTIVITY values (ACTIVE, INACTIVE TIME, etc).
